com.unity.shadergraph/Editor/Data/Graphs/GraphData.cs

```diff
@@ -1562,6 +1624,22 @@ public void OnKeywordChangedNoValidate()
             }
         }
 
+        public void OnDropdownChanged()
+        {
+            OnDropdownChangedNoValidate();
+            ValidateGraph();
+        }
+
+        public void OnDropdownChangedNoValidate()
+        {
+            var allNodes = GetNodes<AbstractMaterialNode>();
```

Comment:
Validate graph should do this already, may be redundant

Reply:
I think in most of those cases, I probably just followed the template used by keywords, but this is possible.

Reply:
eh, Id say leave it. We need to fix our validation/concretization/etc to be smarter anyways.